docs/document.rst

```diff
@@ -592,9 +593,82 @@ For details on **embedded files** refer to Appendix 3.
+
+    :arg bool lossless: include lossless image types (e.g. PNG).
+
+    :arg bool bitonal: include black-and-white images (e.g. FAX).
+
+    :arg bool color: include colored images.
+
+    :arg bool gray: include grayscale images.
+
+    :arg bool set_to_gray: if True, the PDF will be converted to grayscale by executing :meth:`Document.recolor` before all image processing. Please note that this will also change text and vector graphics to grayscale -- not just the images.
+
+    :arg dict options: This parameter is intended for expert users. Except ``set_to_gray``, all other parameters are ignored. It must be an object prepared in the following way: ``options = pymupdf.mupdf.PdfImageRewriterOptions()``. Then attributes of this object can be set to achieve fine-grained control. Following are the adjustable attributes of the ``options`` object and their default (do nothing) values.
+
+    ::
+  
+      options.bitonal_image_recompress_method = 0
+      options.bitonal_image_recompress_quality = None
+      options.bitonal_image_subsample_method = 0
+      options.bitonal_image_subsample_threshold = 0
+      options.bitonal_image_subsample_to = 0
+      options.color_lossless_image_recompress_method = 0
+      options.color_lossless_image_recompress_quality = None
+      options.color_lossless_image_subsample_method = 0
+      options.color_lossless_image_subsample_threshold = 0
+      options.color_lossless_image_subsample_to = 0
+      options.color_lossy_image_recompress_method = 0
+      options.color_lossy_image_recompress_quality = None
+      options.color_lossy_image_subsample_method = 0
+      options.color_lossy_image_subsample_threshold = 0
+      options.color_lossy_image_subsample_to = 0
+      options.gray_lossless_image_recompress_method = 0
+      options.gray_lossless_image_recompress_quality = None
+      options.gray_lossless_image_subsample_method = 0
+      options.gray_lossless_image_subsample_threshold = 0
+      options.gray_lossless_image_subsample_to = 0
+      options.gray_lossy_image_recompress_method = 0
+      options.gray_lossy_image_recompress_quality = None
+      options.gray_lossy_image_subsample_method = 0
+      options.gray_lossy_image_subsample_threshold = 0
+      options.gray_lossy_image_subsample_to = 0
+
+    The ``*_recompress_method`` attributes may be one of the values **0 (never), 1 (same), 2 (lossless), 3 (JPEG), 4 (J2K), 5 (FAX)**. Value 0 will skip this image type altogether and 1 will not change the type. The other values will execute type conversions (as far as technically possible).
+    
+    The ``*_quality`` values are strings of integers from "0" to "100" or ``None``.
+    
+    The ``*_subsample_method`` attributes are either **0 (average)** or **1 (bicubic interpolation)** and refer to how a pixel value is derived from its neighboring pixels during subsampling. For some background see `here <https://en.wikipedia.org/wiki/Bicubic_interpolation>`_.
```

Comment:
Small SEO & accessibility thing - we shouldn't use "here" for links, better to say: 

For some background see [this Wikipedia article about bicubic interpolation](https://en.wikipedia.org/wiki/Bicubic_interpolation).

Reply:
done